include/systems/diff_system.h

```diff
@@ -123,7 +123,8 @@ class DifferentiableSystem : public ImplicitSystem,
    */
   virtual void assembly (bool get_residual,
                          bool get_jacobian,
-                         bool apply_heterogeneous_constraints = false) libmesh_override = 0;
+                         bool apply_heterogeneous_constraints = false,
+			 bool apply_no_constraints = false) libmesh_override = 0;
```

Comment:
Small point: The whitespace isn't aligned here (and in several other cases below).